### ISSUE TITLE
Understanding the `camera_segmentation` dataset.

### ISSUE THREAD

**Issue by aryaman-patel** [waymo-research/waymo-open-dataset#640]
For the panoptic segmentation task, we have the following data: 
```
Index(['key.camera_name', '[CameraImageComponent].image',
       '[CameraSegmentationLabelComponent].panoptic_label_divisor',
       '[CameraSegmentationLabelComponent].panoptic_label',
       '[CameraSegmentationLabelComponent].instance_id_to_global_id_mapping.local_instance_ids',
       '[CameraSegmentationLabelComponent].instance_id_to_global_id_mapping.global_instance_ids',
       '[CameraSegmentationLabelComponent].instance_id_to_global_id_mapping.is_tracked',
       '[CameraSegmentationLabelComponent].sequence_id',
       '[CameraSegmentationLabelComponent].num_cameras_covered'],
      dtype='object')
```
My Questions are:
1. Are the Panoptic labels by default not globally annotated? Because in the 2dpvps tutorial I see a function remapping the instance IDs in each image so that instances for the same object will have the same ID between different cameras and over time.
2. What exactly is the `'num_cameras_covered'` and the `sequence_id` columns for?

**Comment by alexzzhu** [comment]
Hi Aryaman,

The panoptic labels are stored as uint16, which has a max value ~65000.
This puts an upper limit on the number of instances for a single class to
~2000. As a result, we cannot store the global instance IDs in each
panoptic_label, as we may have more than 2000 unique instances in a scene.
Instead, the panoptic labels are unique only for the given frame, and the
mapping from local to global instance ids will provide this mapping to
instance IDs that are globally tracked.

num_cameras_covered indicates the number of cameras that cover or view each
pixel. This is only used for the wSTQ metric used for evaluation. The
sequence_id is consistent for all frames which are tracked, so you can
assume that frames with the same sequence ID have consistent global IDs.
This is mostly used in the training set where we separate sequences into 5
frame segments.

Regards,
Alex

**Comment by shreyas1230** [comment]
A few questions I think might be relevant to this discussion thread:

1. Are the global ids consistent across different sequences (for example, an instance with global_id=5 in sequence 1 also appears in sequence 2 with the same global_id=5)?
2. What is the relation between the `key.segment_context_name` and `sequence_id` columns? From my understanding, the `sequence_id` is the same across frames for a particular 20 second interval. How does this relate to the `segment_context_name`?

**Comment by alexzzhu** [comment]
1. No, there's no guarantees on consistent global ids between different sequence_ids.
2. For the validation and test sets, the sequences span the entire 20 second interval, so the `sequence_id` and `context_name` are consistent. However, for the training set, frames are consistent only over 5-frame intervals (5 frame protos, 25 camera images). Here, the sequence_ids may vary within the same 20 second interval with the same context_name. The goal of the sequence_ids is mostly to identify which frames have consistent global ids in the training set.

**Comment by aryaman-patel** [comment]
The `panoptic labels` are of the type and shape `(uint16) (1280, 1920, 1)` . I was under the impression that it would be in the `rgb` format. 
Just to clarify, the labels here are without the application of the color map? i.e. in order to get it in the rgb format I will have to break it into the segmentation and instance parts and use the function `panoptic_label_to_rgb` from `camera_segmentation_utils`?

**Comment by alexzzhu** [comment]
Yes that's correct. The `panoptic_label` field contains the fusion of the semantic and instance ids for each pixel.
